src/lib/markdown_generators.js

```diff
@@ -0,0 +1,182 @@
+  courseHomeMarkdown += generateCourseCollections(courseData)
+  markdownData.push({
+    name: "_index.md",
+    data: courseHomeMarkdown
+  })
+  let menuIndex = 10
+  courseData["course_pages"].forEach(page => {
+    if (page["text"]) {
+      const pageName = page["short_url"]
+      let courseSectionMarkdown = generateCourseSectionFrontMatter(
+        page["title"],
+        menuIndex
+      )
+      courseSectionMarkdown += generateCourseSectionMarkdown(page, courseData)
+      const sectionData = {
+        name: `sections/${pageName}.md`,
+        data: courseSectionMarkdown
+      }
+      markdownData.push(sectionData)
+      menuIndex += 10
+    }
+  })
+  return markdownData
+}
+
+const generateCourseHomeFrontMatter = courseData => {
+  /*
+      Generate the front matter metadata for the course home page given course_data JSON
+      */
+
+  let courseNumber = courseData["sort_as"]
+  if (courseData["extra_course_number"]) {
+    if (courseData["extra_course_number"]["sort_as_col"]) {
+      courseNumber += ` / ${courseData["extra_course_number"]["sort_as_col"]}`
+    }
+  }
+  const makeTopic = feature => {
+    let topic = ""
+    if (feature["ocw_feature"]) {
+      topic += feature["ocw_feature"]
+    }
+    if (feature["ocw_subfeature"]) {
+      topic += ` - ${feature["ocw_subfeature"]}`
+    }
+    return topic
+  }
+  const frontMatter = {
+    title:              "Course Home",
+    course_title:       courseData["title"],
+    course_image_url:   getCourseImageUrl(courseData),
+    course_description: courseData["description"],
+    course_info:        {
+      instructors: courseData["instructors"].map(
+        instructor =>
+          `Prof. ${instructor["first_name"]} ${instructor["last_name"]}`
+      ),
+      department: titleCase.titleCase(
+        courseData["url"].split("/")[2].replace(/-/g, " ")
+      ),
+      topics:        courseData["course_collections"].map(makeTopic),
+      course_number: courseNumber,
+      term:          `${courseData["from_semester"]} ${courseData["from_year"]}`,
+      level:         courseData["course_level"],
+      menu:          {
+        main: {
+          weight: -10
+        }
+      }
+    }
+  }
+  return `---\n${yaml.safeDump(frontMatter)}---\n`
+}
+
+const generateCourseSectionFrontMatter = (title, menuIndex) => {
+  /*
+      Generate the front matter metadata for a course section given a title and menu index
+      */
+  return `---\n${yaml.safeDump({
+    title: title,
+    menu:  {
+      main: {
+        weight: menuIndex
+      }
+    }
+  })}---\n`
+}
+
+const generateCourseFeatures = courseData => {
+  /*
+      Generate markdown for the "Course Features" section of the home page
+      */
+  let courseFeaturesMarkdown = markdown.headers.hX(5, "Course Features")
+  const courseFeatures = []
+  courseData["course_features"].forEach(courseFeature => {
+    const url = courseFeature["ocw_feature_url"]
+      .replace("/index.htm", "/")
+      .split("/")[-1]
```

Comment:
Negative indexing doesn't work in JS like it does in Python. You will need to do something like `urlPieces[urlPieces.length - 1]`, but maybe there is a better way to do something like this.